src/secrets.ts

```diff
@@ -1,11 +1,29 @@
 import { connect, disconnect } from './db';
 import { prepEnvironment } from './env';
+import { Logger } from './logging';
+import { Time } from './time';
 
-export async function loadEnv(entities: any[] = []) {
+async function loadEnv(entities: any[] = []) {
   await prepEnvironment();
   await connect(entities);
 }
 
-export async function unload() {
+export async function doInDbContext<T>(
+  fn: () => Promise<T>,
+  opts?: { entities?: any[]; logger?: Logger }
+): Promise<T> {
+  const start = Time.now();
+  const logger = opts?.logger ?? Logger.get('MAIN');
+  logger.info(`[RUNNING]`);
+  await loadEnv(opts?.entities ?? []);
+  try {
+    return await fn();
+  } finally {
+    logger.info(`[FINISHED IN ${start.diffFromNow().formatAsDuration()}]`);
+    await unload();
```

Comment:
maybe unload should be before the log line?

Reply:
Debatable, but I personally think it fits better here because unloading takes hardly any time but can theoretically fail by which point the loop itself has finished it's business logic and therefor should log that it's finished.

Anyway it's so obscure. Since I already deployed all the loops to staging and it's so debatable I'd leave it as is.